### DISCUSSION TITLE
Lesson5:  read_value.py vs Brownie console

### DISCUSSION THREAD

**Discussion by CJ-Rose** [smartcontractkit/full-blockchain-solidity-course-py#54] [Q&A] [answered]
I'm getting confused with the switching between local accounts and Rinkeby accounts.  Do we use Brownie console to look at locally deployed contracts, and read_value scripts to look at testnet-deployed contracts? 
Also what happened to Ganache? Did we graduate to Brownie and we don't need it any more, because Brownie also gives us addresses to test with?
I guess I'm trying to make a mental map of all these tools and how they relate to each other... and I can't see the forest for the trees?? I'm googling for visual aides but everything is too generic or it's pictures of brownies.
Thanks for any ideas.

**Comment by PatrickAlphaC** [comment, marked as answer]
Great questions.
> I'm getting confused with the switching between local accounts and Rinkeby accounts.

When we work with a local blockchain, the local blockchain is a ganache chain. When brownie spins up a local blockchain, it gives us some fake accounts to work with. We can use also use `brownie console` to look at contracts on any blockchain. If we look at a testnet with `brownie console --network rinkeby` we are looking at rinkeby. If we just do `brownie console` this will be our "default" chain (which defaults to a local ganache chain)

> Also what happened to Ganache

Whenever we use the `development` network (aka, the default network) we are working with a ganache chain that brownie is secretly running in the background for us!

Mental maps are exactly the right way to think about these, thank you for asking these questions.
